src/core_plugins/vega/public/data_model/es_query_parser.js

```diff
@@ -56,20 +57,41 @@ export class EsQueryParser {
     if (body === undefined) {
       url.body = body = {};
     } else if (!_.isPlainObject(body)) {
-      throw new Error('url.body must be an object');
+      throw new Error(i18n.translate('vega.esQueryParser.urlBodyValueTypeErrorMessage', {
+        defaultMessage: '{configName} must be an object',
+        values: { configName: 'url.body' },
+      }));
     }
 
     // Migrate legacy %context_query% into context & timefield values
     const legacyContext = url[LEGACY_CONTEXT];
     delete url[LEGACY_CONTEXT];
     if (legacyContext !== undefined) {
       if (body.query !== undefined) {
-        throw new Error(`Data url must not have legacy "${LEGACY_CONTEXT}" and "body.query" values at the same time`);
+        throw new Error(i18n.translate('vega.esQueryParser.dataUrlMustNotHaveLegacyAndBodyQueryValuesAtTheSameTimeErrorMessage', {
+          defaultMessage: 'Data {urlParam} must not have legacy "{legacyContext}" and {bodyQueryConfigName} values at the same time',
+          values: {
+            legacyContext: LEGACY_CONTEXT,
+            bodyQueryConfigName: '"body.query"',
+            urlParam: 'url',
+          },
+        }));
       } else if (usesContext) {
-        throw new Error(`Data url must not have "${LEGACY_CONTEXT}" together with "${CONTEXT}" or "${TIMEFIELD}"`);
+        throw new Error(i18n.translate('vega.esQueryParser.dataUrlMustNotHaveLegacyContextTogetherWithContextOrTimefieldErrorMessage', {
+          defaultMessage: 'Data {urlParam} must not have "{legacyContext}" together with "{context}" or "{timefield}"',
+          values: {
+            legacyContext: LEGACY_CONTEXT,
+            context: CONTEXT,
+            timefiels: TIMEFIELD,
+            urlParam: 'url',
+          },
+        }));
       } else if (legacyContext !== true && (typeof legacyContext !== 'string' || legacyContext.length === 0)) {
-        throw new Error(`Legacy "${LEGACY_CONTEXT}" can either be true (ignores time range picker), ` +
-          'or it can be the name of the time field, e.g. "@timestamp"');
+        throw new Error(i18n.translate('vega.esQueryParser.legacyContextCanBeTrueErrorMessage', {
+          // eslint-disable-next-line max-len
+          defaultMessage: 'Legacy "{legacyContext}" can either be {trueValue} (ignores time range picker), or it can be the name of the time field, e.g. {timestampParam}',
```

Comment:
```suggestion
          defaultMessage: 'Legacy {legacyContext} can either be {trueValue} (ignores time range picker), or it can be the name of the time field, e.g. {timestampParam}',
```